File: frontend/lib/models/conversation.dart
class Conversation {
  final String id;
  final String boutiqueId;
  final String customerName;
  final String customerEmail;
  final String? customerPhone;
  final String? lastMessageAt;
  final String? lastMessagePreview;
  final int unreadCount;
  final String? createdAt;

  Conversation({
    required this.id,
    required this.boutiqueId,
    required this.customerName,
    required this.customerEmail,
    this.customerPhone,
    this.lastMessageAt,
    this.lastMessagePreview,
    this.unreadCount = 0,
    this.createdAt,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) => Conversation(
    id: json['id'].toString(),
    boutiqueId: json['boutiqueId'].toString(),
    customerName: json['customerName'] ?? '',
    customerEmail: json['customerEmail'] ?? '',
    customerPhone: json['customerPhone'],
    lastMessageAt: json['lastMessageAt'],
    lastMessagePreview: json['lastMessagePreview'],
    unreadCount: json['unreadCount'] ?? 0,
    createdAt: json['createdAt'],
  );
}

File: backend/src/main/java/io/makewebsite/entity/Conversation.java
package io.makewebsite.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "conversations")
public class Conversation {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(updatable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "boutique_id", nullable = false)
    private Boutique boutique;

    @NotNull
    @Size(max = 150)
    @Column(name = "customer_name", nullable = false)
    private String customerName;

    @NotNull
    @Size(max = 150)
    @Column(name = "customer_email", nullable = false)
    private String customerEmail;

    @Size(max = 20)
    @Column(name = "customer_phone")
    private String customerPhone;

    @Column(name = "last_message_at", nullable = false)
    private LocalDateTime lastMessageAt;

    @Column(name = "last_message_preview", columnDefinition = "TEXT")
    private String lastMessagePreview;

    @Builder.Default
    @Column(name = "unread_count")
    private Integer unreadCount = 0;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (lastMessageAt == null) lastMessageAt = LocalDateTime.now();
    }
}
